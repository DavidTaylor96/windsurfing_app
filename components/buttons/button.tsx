import React, { FC } from 'react';
import { Pressable } from 'react-native';
import useDefaultSafeView from '../../hooks/useDefaultSafeView';

interface IButtonProps {
  onPress: () => void;
}

export const Button: FC<IButtonProps> = ({ children, onPress }) => {

  const inset = useDefaultSafeView();

  return (
    <Pressable
      style={{
        position: 'absolute',
        top: inset.top,
        right: 12,
        padding: 8,
        borderRadius: 8,
        backgroundColor: 'white',
      }}
      onPress={onPress}
    >
      {children}
    </Pressable>
  );
};
