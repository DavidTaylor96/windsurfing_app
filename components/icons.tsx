import { Feather } from '@expo/vector-icons';
import React, { FC } from 'react';

export type IconType = keyof typeof Feather.glyphMap;
export interface IIconProps {
  icon: IconType;
  color?: string;
  size?: number;
}

export const Icon: FC<IIconProps> = (props) => {
  return (
    <Feather
      name={props.icon}
      size={props.size ? props.size : 24}
      color={props.color ? props.color : 'black'}
    />
  );
};
