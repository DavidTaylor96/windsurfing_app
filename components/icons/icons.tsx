import { Feather } from '@expo/vector-icons';
import React, { FC } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

export type IconType = keyof typeof Feather.glyphMap;
export interface IIconProps {
  icon: IconType;
  color?: string;
  size?: number;
  style?: StyleProp<ViewStyle>;
}

export const Icon: FC<IIconProps> = (props) => {
  return (
    <Feather
      name={props.icon}
      size={props.size ? props.size : 24}
      color={props.color ? props.color : 'black'}
      style={props.style}
    />
  );
};
