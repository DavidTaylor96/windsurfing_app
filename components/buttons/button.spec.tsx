import * as React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import renderer from 'react-test-renderer';
import { Icon } from '../icons';
import { Button } from './button';

it(`Button`, () => {
  const mockFunction = jest.fn();

  const button = renderer.create(
    <SafeAreaProvider>
      <Button onPress={mockFunction}>
        <Icon icon="navigation" />
      </Button>
    </SafeAreaProvider>
  );

  button.root.props.children.props.onPress();

  expect(button).toMatchSnapshot();

  expect(mockFunction).toHaveBeenCalledTimes(1);
});
