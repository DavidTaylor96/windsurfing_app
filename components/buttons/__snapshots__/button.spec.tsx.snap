// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 8,
      "padding": 8,
      "position": "absolute",
      "right": 12,
      "top": 15,
    }
  }
>
  <Text />
</View>
`;
